src/utils.jl

```diff
@@ -40,9 +43,15 @@ Sample `k` element from unit range `r` without repetition and eventually excludi
 ### Implementation Notes
 Unlike [`sample!`](@ref), does not produce side effects.
 """
-sample(a::AbstractVector, k::Integer; exclude=()) = sample!(getRNG(), collect(a), k; exclude=exclude)
-
-getRNG(seed::Integer=-1) = seed >= 0 ? MersenneTwister(seed) : GLOBAL_RNG
+sample(
+    a::AbstractVector, k::Integer;
+    exclude=(), rng::Union{Nothing, AbstractRNG}=nothing, seed::Union{Nothing, Integer}=-1
+) = sample!(rng_from_rng_or_seed(rng, seed), collect(a), k; exclude=exclude)
+
+function getRNG(seed::Integer=-1)
+    seed >= 0 && seed!(GLOBAL_RNG, seed)
```

Comment:
What is the reason for seeding the `GLOBAL_RNG` instead of returning new random generator? This might lead to surprising as specifying a non-negative integer now changes the state of `GLOBAL_RNG` instead of creating a new independent one as before.

Reply:
No reason, this is a leftover from the original PR. I will restore that to what was in master.

Reply:
Fixed.